syz-cluster/pkg/controller/api_test.go

```diff
@@ -70,17 +72,52 @@ func TestAPISaveFinding(t *testing.T) {
 		finding := &api.NewFinding{
 			SessionID:    ids.SessionID,
 			TestName:     "test",
+			Title:        "title",
 			Report:       []byte("report"),
 			Log:          []byte("log"),
 			SyzRepro:     []byte("syz repro"),
 			SyzReproOpts: []byte("syz_repro_opts"),
-			CRepro:       []byte("C repro"),
 		}
 		err = client.UploadFinding(ctx, finding)
 		assert.NoError(t, err)
-		// Even if the finding is reported the second time, it must still not fail.
+		// Even if the same finding is reported the second time, it must still not fail.
+		err = client.UploadFinding(ctx, finding)
+		assert.NoError(t, err)
+	})
+
+	t.Run("add C repro", func(t *testing.T) {
+		finding := &api.NewFinding{
+			SessionID:    ids.SessionID,
+			TestName:     "test",
+			Title:        "title",
+			Report:       []byte("report"),
+			Log:          []byte("log"),
+			SyzRepro:     []byte("syz repro"),
+			SyzReproOpts: []byte("syz_repro_opts"),
+			CRepro:       []byte("C repro"),
+		}
 		err = client.UploadFinding(ctx, finding)
 		assert.NoError(t, err)
+		// Verify that C repro has appeared indeed.
+		findingRepo := db.NewFindingRepository(env.Spanner)
+		findings, err := findingRepo.ListForSession(ctx, ids.SessionID, db.NoLimit)
+		require.NoError(t, err)
+		require.Len(t, findings, 1)
+		assert.NotEmpty(t, findings[0].CReproURI)
+	})
+
+	t.Run("session stopped", func(t *testing.T) {
```

Comment:
There are places in syzkaller where small unit tests eventually evolved into integration tests.
This function looks now like one more example of this problem.
I think every function deserves a set of unit tests.
For some set of functions we need integration tests.
WDYT?

Reply:
> There are places in syzkaller where small unit tests eventually evolved into integration tests.

I guess it's mostly because there was no choice left and I don't think it's necessarily bad  :) 

Our code has to support multiple complex scenarios, and if we are able to ensure that all these succeed, it's probably just not worth it to write unit tests for all the pieces of code involved. Especially in the cases like these where we interact with the DB a lot. Writing unit tests would be quite painful here and they likely won't really improve our testing on top of the integration test.